test/e2e/scoped_client_test.go

```diff
@@ -61,7 +61,7 @@ var _ = Describe("Scoped Client", func() {
 			// lack of permission.
 			name: "ServiceAccountDoesNotHaveAnyPermission",
 			assertFunc: func(errGot error) {
-				require.True(GinkgoT(), k8serrors.IsForbidden(errGot))
+				Expect(k8serrors.IsForbidden(errGot)).To(BeTrue())
```

Comment:
Side note: The api-machinery error classification utils are used throughout the end-to-end tests. It might be nice to have a matcher that accepts a `func(error) bool`, and tries to use reflection to grab the function's name (e.g. `IsForbidden`) to generate meaningful failure messages. It's not a necessity, but it would be nice to have better messages than "expected true but was false" without having to write them all manually.

Reply:
Are you referring to writing a [Custom Matcher](https://onsi.github.io/gomega/#adding-your-own-matchers). If so, could this be done as part of a dedicated PR for this specific change? I'll make a note of this and work on it on the side.

Reply:
Being done as part of #1730